doc/GUIDE.md

```diff
@@ -305,21 +223,30 @@ value here says to use [LTS Haskell version
 `stack setup` installs that version of GHC). There are a number of values you
 can use for `resolver`, which we'll cover later.
 
-The final file of import is `helloworld.cabal`. stack is built on top of the
-Cabal build system. In Cabal, we have individual *packages*, each of which
-contains a single `.cabal` file. The `.cabal` file can define 1 or more
-*components*: a library, executables, test suites, and benchmarks. It also
-specifies additional information such as library dependencies, default language
+The files of final importance are the `helloworld.cabal` and `package.yaml`.
+
+Since Stack 1.6.1, the `package.yaml` is an alternative package format that is
+provided built-in by stack through [the hpack tool](https://github.com/sol/hpack). The
+default behaviour is to generate the `.cabal` file from this `package.yaml`, and you
+can see that from the general message header in your new `helloworld.cabal` file. It
+is therefore important to understand that your `package.yaml` drives the package configuration.
+
```

Comment:
Do you think it would be a good idea around here to explicitly mention that they usually shouldn't be modifying the .cabal file? It's certainly strongly implied, but those who aren't familiar with hpack might still not realize that exactly.